ms2deepscore/MetadataFeatureGenerator.py

```diff
@@ -1,34 +1,110 @@
 
-class PrecursorMZFeatureGenerator(MetadataFeatureGenerator):
-    """Class for generating feature from precursor-m/z (here: simply m/z divided by 1000).
-    """
-    def generate_features(self) -> float:
-        precursor_mz = self.metadata.get("precursor_mz")
-        assert precursor_mz is not None, "No precursor mz was found, preprocess your spectra first using matchms"
-        return precursor_mz/1000
+    @classmethod
+    def load_from_dict(cls, json_dict: dict):
+        """This method should be implemented by child classes Create class instance from json.
+        """
+        raise NotImplementedError
 
+    def __eq__(self, other):
+        if isinstance(other, type(self)):
+            return self.__dict__ == other.__dict__
+        return False
 
-class IonizationModeFeatureGenerator(MetadataFeatureGenerator):
-    """Class for generating feature from ionization mode. Here simply 0: positive mode and 1: negative mode.
-    """
-    def generate_features(self) -> float:
-        ionization_mode = self.metadata.get("ionization_mode")
-        if ionization_mode == "positive":
-            return 0
-        if ionization_mode == "negative":
+
+class StandardScaler(MetadataFeatureGenerator):
+    def __init__(self, metadata_field: str, mean: float, std: float = None):
+        self.metadata_field = metadata_field
+        self.mean = mean
+        self.standard_deviation = std
+
+    def generate_features(self, metadata: Metadata):
+        feature = metadata.get(self.metadata_field, None)
+        assert self.metadata_field is not None, f"Metadata entry for {self.metadata_field} is missing."
+        assert isinstance(feature, (int, float))
+        if self.standard_deviation:
+            return (feature - self.mean) / self.standard_deviation
+        return feature - self.mean
+
+    @classmethod
+    def load_from_dict(cls, json_dict: dict):
+        """Create StandardScaler instance from json.
+        """
+        return cls(json_dict["metadata_field"],
+                   json_dict["mean"],
+                   json_dict["standard_deviation"],)
+
+
+class OneHotEncode(MetadataFeatureGenerator):
+    def __init__(self, metadata_field: str,
+                 entries_becoming_one):
+        self.metadata_field = metadata_field
+        self.entries_becoming_one = entries_becoming_one
+
+    def generate_features(self, metadata: Metadata):
+        feature = metadata.get(self.metadata_field, None)
+        assert self.metadata_field is not None, f"Metadata entry for {self.metadata_field} is missing."
+        if feature == self.entries_becoming_one:
+            return 1
+        return 0
+
+    @classmethod
+    def load_from_dict(cls, json_dict: dict):
+        """Create StandardScaler instance from json.
+        """
+        return cls(json_dict["metadata_field"],
+                   json_dict["entries_becoming_one"])
+
+
+class FeatureToBinary(MetadataFeatureGenerator):
+    def __init__(self, metadata_field: str,
```

Comment:
```suggestion
class CategoricalToBinary(MetadataFeatureGenerator):
    """Converts categorical features (e.g. strings) into a binary 1 or 0 feature.
    """
    def __init__(self, metadata_field: str,
```

Reply:
I would distinguish between categorical and numerical (or similar). Because it would also make sense to have another version that can create binary features from numbers, say "large molecules" --> 1 for m/z > 500 .